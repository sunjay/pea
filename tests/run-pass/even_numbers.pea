fn print_even(start: i64, end: i64, offset: i64) {
    let i = start;
    while i <= end {
        let value = i;
        i += offset;

        if value % 2 != 0 {
            continue;
        }

        println!(value);
    }
}

fn main() {
    println!(b"Even numbers between 0 and 10, going up by 1");
    print_even(0, 10, 1);
    println!(b"Even numbers between 0 and 12, going up by 3");
    print_even(0, 12, 3);
}
