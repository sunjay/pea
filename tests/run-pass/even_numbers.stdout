Even numbers between 0 and 10, going up by 1
0
2
4
6
8
10
Even numbers between 0 and 12, going up by 3
0
6
12
